examples/apps/quizmaster_cozmo.py

```diff
@@ -0,0 +1,403 @@
+        num_valid_players = 0
+        for player in self._players:
+            if player.verify_setup():
+                num_valid_players += 1
+        return (num_valid_players > 0)
+
+    def get_player_for_object_id(self, object_id):
+        for player in self._players:
+            if player.object_id == object_id:
+                return player
+        cozmo.logger.warn("No player for object_id %s", object_id)
+        return None
+
+    def on_cube_tapped(self, evt, **kw):
+        # find the player for that cube, and handle the tap as appropriate
+        player = self.get_player_for_object_id(evt.obj.object_id)
+        if self._buzzing_in_accepted and self._answering_player is None:
+            if player and not player.has_buzzed_in:
+                self._answering_player = player
+                player.on_buzzed_in()
+        elif self._answers_accepted and player == self._answering_player:
+            if player:
+                player.cycle_answer()
+
+    def turn_player_lights_on(self):
+        for player in self._players:
+            player.turn_light_on()
+
+    def turn_player_lights_off(self):
+        for player in self._players:
+            player.turn_light_off()
+
+    def get_next_question(self) -> QuizQuestion:
+        if len(self._questions) > 0:
+            i = randrange(len(self._questions))
+            question = self._questions.pop(i)
+            return question
+        else:
+            print("Out of questions!")
+            return None
+
+    def create_option_string(self, list_of_options) -> str:
+        # Build a string that lists all of the options in order.
+        text = "Is it "
+        for i in range(len(list_of_options)):
+            conjunction = ""
+            if i > 0:
+                is_last_option = (i == (len(list_of_options) - 1))
+                conjunction = " or " if is_last_option else ", "
+            text += conjunction + str(i+1) + ": " + list_of_options[i]
+        return text
+
+    def say_text(self, text, in_parallel=False):
+        print("%s" % text)
+        return self._robot.say_text(text, in_parallel=in_parallel)
+
+    async def wait_for_answer(self, player):
+        # Wait for player's answer (whatever they leave selected after x seconds)
+        # This is after Cozmo has finished speaking, so we've already given the
+        # player a few seconds.
+        await asyncio.sleep(2.0)
+        return player.answer_number
+
+    def get_winning_players(self):
+        # get a list of all the players with the top score
+        winning_players = []
+        for player in self._players:
+            if len(winning_players) == 0 or player.score > winning_players[0].score:
+                winning_players = [player]
+            elif player.score == winning_players[0].score:
+                winning_players.append(player)
+        return winning_players
+
+    async def report_leader(self, is_final_score):
+        # report however is currently winning / won
```

Comment:
Could rewrite as "Report the winning player" or similar. "however" should be "whoever" if original text is kept.